packages/x-tree-view/src/internals/plugins/useTreeViewSelection/useTreeViewSelection.ts

```diff
@@ -88,16 +88,16 @@ export const useTreeViewSelection: TreeViewPlugin<UseTreeViewSelectionSignature>
     if (keepExistingSelection) {
       const cleanSelectedItems = convertSelectedItemsToArray(models.selectedItems.value);
       const isSelectedBefore = instance.isItemSelected(itemId);
-      if (isSelectedBefore && (newValue === false || newValue == null)) {
+      if (isSelectedBefore && (isSelected === false || isSelected == null)) {
         newSelected = cleanSelectedItems.filter((id) => id !== itemId);
-      } else if (!isSelectedBefore && (newValue === true || newValue == null)) {
+      } else if (!isSelectedBefore && (isSelected === true || isSelected == null)) {
         newSelected = [itemId].concat(cleanSelectedItems);
       } else {
         newSelected = cleanSelectedItems;
       }
     } else {
       // eslint-disable-next-line no-lonely-if
-      if (newValue === false) {
+      if (isSelected === false || (isSelected == null && instance.isItemSelected(itemId))) {
```

Comment:
Small fix in a scenario that was never used before (single selection, `isSelected` not defined and the item is selected)